### PR TITLE
fix(search): add defaultCloseButtonLabel

### PR DESCRIPTION
**Pull request by jendowns**
Addresses #3454 (note: I'm not setting this to auto close so that @dakahn has a chance to confirm)

#### Changelog

**Changed**

- set default prop value for `CloseButtonLabel` (as it should never be blank, but making it `isRequired` would be a breaking change)
- set a default knob value for `closeButtonLabel` in Storybook (so as not to generate DAP noise & show best practices to the user from the start)
- update snapshots (NOTE: please see my question here about `aria-label={true}` on the icon - https://github.com/carbon-design-system/carbon/pull/3483#discussion_r305919306)

### CODE REVIEW

File: packages/react/src/components/DataTable/__tests__/__snapshots__/DataTable-test.js.snap [hunk 1]
```diff
@@ -2319,7 +2323,7 @@ exports[`DataTable should render 1`] = `
                             xmlns="http://www.w3.org/2000/svg"
                           >
                             <svg
-                              aria-hidden={true}
+                              aria-label="Clear search query"
```
Comment (jendowns) [inline packages/react/src/components/DataTable/__tests__/__snapshots__/DataTable-test.js.snap:2326]: @dakahn Shouldn't this stay `aria-hidden={true}` here for the icon? 🤔

**Comment by jnm2377** [comment]
Hi @jendowns thanks for tackling this issue! It looks like @emyarod is including this in another PR. https://github.com/carbon-design-system/carbon/pull/3365

File: packages/react/src/components/DataTable/__tests__/__snapshots__/DataTable-test.js.snap (hunk 1 above)
Comment (dakahn) [reply packages/react/src/components/DataTable/__tests__/__snapshots__/DataTable-test.js.snap:2326]: If the SVG purely decorative, but I think in this case, it's a button, right?

Comment (jendowns) [reply packages/react/src/components/DataTable/__tests__/__snapshots__/DataTable-test.js.snap:2326]: @dakahn for sure but wouldn't this be redundant if parent button element already has the same `aria-label`? 🤔

Comment (dakahn) [reply packages/react/src/components/DataTable/__tests__/__snapshots__/DataTable-test.js.snap:2326]: Oh, you're right! Didn't catch that with the snippet above. My bad.

**Pull request closed without merging**